Utilities/DB.php

```diff
@@ -3,5 +3,141 @@
+     */
+    public function fillCatBreedToDB()
+    {
+        $dbconnect = $this->dbConnect();
+        $catBreeds = $this->getCatBreed();
+        foreach($catBreeds as $id=>$breed) {
+            $sql = $dbconnect->prepare('INSERT INTO `breed` (breed) VALUES (\'' . $breed . '\');');
+            $sql->execute();
+        }
+    }
+
+    /**
+     * Get the return value of getCatBreed() method
+     * Use the breed id to use it in the API url to send the request
+     * Make the response into an array
+     * Return an associative array of breed name as the $key and all the image source url belongs to that breed as $value(which is in array format)
+     * @return array
+     */
+    public function getCatImg()
+    {
+        $catBreeds = $this->getCatBreed();
+        $imgSrcArray = [];
+        foreach ($catBreeds as $id=>$name) {
+            $catImgApiUrl = 'https://api.thecatapi.com/v1/images/search?breed_ids=' . $id . '&limit=21';
+            $curl = curl_init();
+            curl_setopt_array($curl, array(
+                CURLOPT_URL => $catImgApiUrl,
+                CURLOPT_CUSTOMREQUEST => "GET",
+                CURLOPT_RETURNTRANSFER => true,
+                CURLOPT_HTTPHEADER => array(
+                    'x-api-key: ec254e44-3996-458b-8522-4933954d8fcd'
+                ),
+            ));
+
+            $imgApiResponse = curl_exec($curl);
+            $error = curl_error($curl);
+            curl_close($curl);
+
+            $responseArray = json_decode($imgApiResponse, true);
+            $breedImgs = [];
+
+            foreach($responseArray as $item) {
+                array_push($breedImgs, $item["url"]);
+            }
+            $imgSrcArray[$name] = $breedImgs;
+        }
+        return $imgSrcArray;
+    }
+
+    /**
+     * Get DB connection from the dbConnect() method
+     * Get the return value from getCatImg() method
+     * Get the return value of getCatBreed() method, turn it into an indexed array
+     * Create a new array $sqlArray which is the array of strings consist of 'img url($url)' and 'breed id($breedID)' that will be pushed to database directly as is
+     * @return voide
+     */
+    public function fillCatImg()
+    {
+        $dbconnect = $this->dbConnect();
+        $catImgSrcArray = $this->getCatImg();
+        $catBreedArray = $this->getCatBreed();
+        $catBreedIndexedArray = [];
+
+        foreach($catBreedArray as $breed) {
+            $catBreedIndexedArray[] = $breed;
+        }
+
+        $sqlArray = [];
+
+        for($breedIndex = 0; $breedIndex < count($catImgSrcArray); $breedIndex++) {
+            foreach($catImgSrcArray[$catBreedIndexedArray[$breedIndex]] as $url) {
+                $breedID = $breedIndex + 1;
+                $sqlArray[] = "('$url', $breedID)";
+            }
+        }
+
+        $sqlString = implode(',', $sqlArray);
+
+        $sql = $dbconnect->prepare('INSERT INTO `img` (img_src, breed_id) VALUES ' . $sqlString . ';');
+        $sql->execute();
+    }
```

Comment:
The actual method calls/data flow should be be included
The calls to other methods within methods should be tidied

Reply:
yeah I agree. This should be done in s2t3 branch I believe